tensorflow_addons/layers/tcn.py

```diff
@@ -0,0 +1,336 @@
+            For example "name=str", Name of the model.
+            Use unique names when using multiple TCN.
+    Returns:
+        A TCN layer.
+    """
+
+    def __init__(self,
+                 filters=64,
+                 kernel_size=2,
+                 stacks=1,
+                 dilations=[1, 2, 4, 8, 16, 32, 64],
+                 padding='causal',
+                 use_skip_connections=True,
+                 dropout_rate=0.0,
+                 return_sequences=False,
+                 activation='linear',
+                 kernel_initializer='he_normal',
+                 use_batch_norm=False,
+                 **kwargs):
+
+        super(TCN, self).__init__(**kwargs)
+
+        self.return_sequences = return_sequences
+        self.dropout_rate = dropout_rate
+        self.use_skip_connections = use_skip_connections
+        self.stacks = stacks
+        self.kernel_size = kernel_size
+        self.filters = filters
+        self.dilations = dilations
+        self.activation = activation
+        self.padding = padding
+        self.kernel_initializer = kernel_initializer
+        self.use_batch_norm = use_batch_norm
+
+        # validate paddings
+        validate_paddings = ['causal', 'same']
+        if padding not in validate_paddings:
+            raise ValueError(
+                "Only 'causal' or 'same' padding are compatible for this layer"
+            )
+
+        self.main_conv1D = tf.keras.layers.Conv1D(
+            filters=self.filters,
+            kernel_size=1,
+            padding=self.padding,
+            kernel_initializer=self.kernel_initializer)
+
+        # list to hold all the member ResidualBlocks
+        self.residual_blocks = list()
+        total_num_blocks = self.stacks * len(self.dilations)
+        if not self.use_skip_connections:
+            total_num_blocks += 1  # cheap way to do a false case for below
+
+        for _ in range(self.stacks):
+            for d in self.dilations:
+                self.residual_blocks.append(
+                    ResidualBlock(
+                        dilation_rate=d,
+                        filters=self.filters,
+                        kernel_size=self.kernel_size,
+                        padding=self.padding,
+                        activation=self.activation,
+                        dropout_rate=self.dropout_rate,
+                        use_batch_norm=self.use_batch_norm,
+                        kernel_initializer=self.kernel_initializer,
+                        last_block=len(self.residual_blocks) +
+                        1 == total_num_blocks,
+                        name='residual_block_{}'.format(
+                            len(self.residual_blocks))))
+
+        # this is done to force keras to add the layers in the list to
+        # self._layers
+        for layer in self.residual_blocks:
+            self.__setattr__(layer.name, layer)
+
+        self.lambda_layer = tf.keras.layers.Lambda(lambda tt: tt[:, -1, :])
```

Comment:
1. Use a more descriptive name than `lambda_layer`. 
2. Should be created only when `return_sequences=True`.

Reply:
got it, thanks @Squadrick , will update!

Reply:
updated!